### ISSUE TITLE
faster and better random numbers?

### ISSUE THREAD

**Issue by Abalieno** [CleverRaven/Cataclysm-DDA#5249]
Consider I'm a total noob, but maybe I got something nice.

I've noticed there was a discussion in the past about rand and fastrand, but right now you still use the standard rand() right?

Well, I've fiddled a bit with my own code and made some approximate benchmarks on C++11.

Without optimization rand is twice as fast as the C++11 implementation of Mersenne or minstd_rand0.

But with -O3 I've noticed that both Mersenne and the other become FASTER than rand, and would likely provide much better results too.

So this is really just two lines of code, if we compile with C++11.

```C+++
#include <random>

long rng(long low, long high)
{
    static std::mt19937 rnumber; // uses Mersenne Twister
    std::uniform_int_distribution<unsigned> u(low, high);
    return u(rnumber);
}
```

**Comment by DLaboratory** [comment]
I implemented `xorshift128+` RNG in #17383. It's a better RNG algorithm than MT19937 for gaming.

**Comment by mugling** [comment]
Conclusion from #17383 was that any reimplementation needs to be backed by empiric profiling data
